sys/vm_map.c

```diff
@@ -132,10 +137,57 @@ void vm_map_protect(vm_map_t *map, vm_addr_t start, vm_addr_t end,
                     vm_prot_t prot) {
 }
 
+int vm_map_findspace(vm_map_t *map, vm_addr_t start, size_t length,
+                     vm_addr_t /*out*/ *addr) {
+
+  assert(is_aligned(length, PAGESIZE));
+  /* Bounds check */
+  if (start < map->pmap->start)
+    start = map->pmap->start;
+  if (start + length >= map->pmap->end)
```

Comment:
Looks like this code will never allocate the last page.
Shouldn't you check for `(start + length > map->pmap->end)` here?